pkg/labels/http_applicator.go

```diff
@@ -313,6 +314,22 @@ func (h *httpApplicator) getMatches(selector labels.Selector, labelType Type, ca
 	return labeled, nil
 }
 
+// The caller is responsible for closing the resp.Body
```

Comment:
you may as well `defer resp.Body.Close()` in here because the caller doesn't even use the body (except to close it)

Reply:
Yeah you're right. Also you made me notice that the `resp.Body.Close()` call on 270 would panic because `resp` is nil in that case.